### PR TITLE
Add unit tests for build URL and app authentication

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
### Proposed changes in this pull request
Add unit tests for following methods in `application.authentication.framework.util.FrameworkUtils`
```
appendQueryParamsToUrl
buildURLWithQueryParams
GetAppAuthenticatorByName
```

### CODE REVIEW

**Review by Thumimku — APPROVED** [review]
Looks Good To Me (LGTM)

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 1]
```diff
@@ -20,24 +20,38 @@
...
-@PrepareForTest({ConfigurationFacade.class})
-public class FrameworkUtilsTest {
+@WithCarbonHome
+@PrepareForTest({ConfigurationFacade.class, FrameworkServiceComponent.class})
+public class FrameworkUtilsTest extends PowerMockIdentityBaseTest {
 
     @Mock
     ConfigurationFacade mockedConfigurationFacade;
 
+
```
Comment (sumedhe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:50]: Unnecessary new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java [hunk 2]
```diff
@@ -48,6 +62,37 @@ private void setMockedConfigurationFacade() {
         when(ConfigurationFacade.getInstance()).thenReturn(mockedConfigurationFacade);
     }
 
+    private void mockFrameworkServiceComponent() {
```
Comment (sumedhe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:65]: Add a new line after the method signature.
Fix in other relevant places.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:50]: Already fixed with commit baf49d4b0e85432467ae5848dcf443c0032ce5c9 in https://github.com/wso2/carbon-identity-framework/pull/3509 PR

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtilsTest.java:65]: Already fixed with commit baf49d4b0e85432467ae5848dcf443c0032ce5c9 in https://github.com/wso2/carbon-identity-framework/pull/3509 PR

**Pull request merged**